django/http/response.py

```diff
@@ -383,7 +384,10 @@ def streaming_content(self):
     @streaming_content.setter
     def streaming_content(self, value):
         # Ensure we can never iterate on "value" more than once.
-        self._iterator = iter(value)
+        if hasattr(value, 'read'):
```

Comment:
It seems a bit weird to add this functionality on `StreamingHtttpResponse` in 1.7 and then remove it in 1.8 -- unless you plan to forward port this change?